internal/provider/kubernetes/gatewayclass.go

```diff
@@ -30,26 +30,26 @@ type gatewayClassReconciler struct {
 	log           logr.Logger
 
 	initializeOnce sync.Once
-	resourceTable  *ResourceTable
+	resources      *message.ProviderResources
 }
 
 // newGatewayClassController creates the gatewayclass controller. The controller
 // will be pre-configured to watch for cluster-scoped GatewayClass objects with
 // a controller field that matches name.
-func newGatewayClassController(mgr manager.Manager, cfg *config.Server, resourceTable *ResourceTable) error {
+func newGatewayClassController(name string, mgr manager.Manager, logger logr.Logger, resources *message.ProviderResources) error {
 	cli := mgr.GetClient()
-	uh := status.NewUpdateHandler(cfg.Logger, cli)
+	uh := status.NewUpdateHandler(logger, cli)
 	if err := mgr.Add(uh); err != nil {
 		return fmt.Errorf("failed to add status update handler %v", err)
 	}
 
-	resourceTable.Initialized.Add(1)
+	resources.Initialized.Add(1)
 	r := &gatewayClassReconciler{
 		client:        cli,
-		controller:    gwapiv1b1.GatewayController(cfg.EnvoyGateway.Gateway.ControllerName),
+		controller:    gwapiv1b1.GatewayController(name),
```

Comment:
The value of `controller` should continue to be exposed to users to support running multiple Envoy Gateways in a cluster.

Reply:
not sure what you are referring to here